tests/Microsoft.Identity.Test.Integration.netcore/HeadlessTests/LegacyPopTest.NetFwk.cs

```diff
@@ -20,6 +21,51 @@
 
 namespace Microsoft.Identity.Test.Integration.HeadlessTests
 {
+    internal static class MsAuth10AtPop
+    {
+        internal class MsAuth10AtPopOperation : IAuthenticationOperation
+        {
+            private readonly string _reqCnf;
+
+            public MsAuth10AtPopOperation(string keyId, string reqCnf)
+            {
+                KeyId = keyId;
+                _reqCnf = reqCnf;
+            }
+            public int TelemetryTokenType => 4; // as per TelemetryTokenTypeConstants
+
+            public string AuthorizationHeaderPrefix => "Bearer"; // these tokens go over bearer
+
+            public string KeyId { get; }
+
+            public string AccessTokenType => "pop"; // eSTS returns token_type=pop and MSAL needs to know
+
+            public void FormatResult(AuthenticationResult authenticationResult)
+            {
+                // no-op, adding the SHR is done by the caller
+            }
+
+            public IReadOnlyDictionary<string, string> GetTokenRequestParams()
+            {
+                return new Dictionary<string, string>()
+                {
+                    {"req_cnf", Base64UrlEncoder.Encode(_reqCnf) },
+                    {"token_type", "pop" }
+                };
+            }
+        }
+
+        internal static AcquireTokenForClientParameterBuilder WithAtPop(
```

Comment:
@sruke - this is the Id.Web extension. Instead of using `OnBefore...` and `WithKeyID` you use `WithAuthenticationOperation`

Rest of the code is a test.

Reply:
While trying to use WithAuthenticationOperation, I discovered that this method is internal to MSAL. Can it be made public for use in Identity.Web?

Reply:
Updated the test to use the public API `WithAuthenticationExtension`